java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1091,6 +1111,8 @@ public Scalar reduce(ReductionOp op) {
    * {@link Scalar#isValid()} method of the result will return false.
    */
   public Scalar reduce(ReductionOp op, DType outType) {
+    if(op == ReductionOp.STD && outType != DType.FLOAT32)
+      outType = DType.FLOAT64;
```

Comment:
If they passed in a specific output type we should honor it.  If it is not going to work, then we should have an assertion.

Reply:
So a user using this direct reduce method (over the standardDeviation method) directly that does not specify a valid type should receive an error from the java side before any actual execution?
```
    if(op == ReductionOp.STD)
      assert (outType == DType.FLOAT64 || outType == DType.FLOAT32);
```

Reply:
If the native side returns an error message that a user could use to debug it, then I would prefer to have the native side handle it, but if the error message is not clear, or if it panics then we should have the assertion.

Reply:
Native error is useful and informative enough to debug it.

Abbreviated error trace
```
ai.rapids.cudf.CudfException: cuDF failure at: ../cudf/cpp/src/reductions/compound.cuh:114: Unsupported output data type
	at ai.rapids.cudf.Cudf.reduce(Native Method)
	at ai.rapids.cudf.Cudf.reduce(Cudf.java:121)
	at ai.rapids.cudf.ColumnVector.reduce(ColumnVector.java:1116)
	at ai.rapids.cudf.ColumnVector.reduce(ColumnVector.java:1100)
```